client/src/ui/molecules/paginator/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { Link } from "react-router-dom";
+import { range } from "../../../utils";
+
+function PageLink({ page }: { page: number }) {
+  return <Link to={`?page=${page}`}>{page}</Link>;
+}
```

Comment:
Did you check whether you could use `children` with `page` as fallback here?

```suggestion
function PageLink({ children, page }: { children: React.ReactNode, page: number }) {
  return <Link to={`?page=${page}`}>{children || page}</Link>;
}
```